frontend/src/App.js

```diff
@@ -12,66 +11,80 @@ class App extends Component {
     super(props);
     this.state = {
       sql: `SELECT COUNT(*) as num_commits, month, repo_id, committer_email FROM (
-    SELECT MONTH(committer_when) as month, r.id as repo_id, committer_email
-    FROM repositories r
-    INNER JOIN refs ON refs.repository_id = r.id AND refs.name = 'HEAD'
-    INNER JOIN commits c ON YEAR(committer_when) = 2018 AND history_idx(refs.hash, c.hash) >= 0
+  SELECT MONTH(committer_when) as month, r.repository_id as repo_id, committer_email
+  FROM repositories r
+  INNER JOIN refs ON refs.repository_id = r.repository_id AND refs.ref_name = 'HEAD'
+  INNER JOIN commits c ON YEAR(committer_when) = 2018 AND history_idx(refs.commit_hash, c.commit_hash) >= 0
 ) as t
 GROUP BY committer_email, month, repo_id`,
-      response: undefined,
-      loading: false,
-      error: undefined
+      results: new Map()
     };
 
     this.handleTextChange = this.handleTextChange.bind(this);
     this.handleSubmit = this.handleSubmit.bind(this);
+    this.handleRemoveResult = this.handleRemoveResult.bind(this);
+
+    this.uniqueKey = 0;
   }
 
   handleTextChange(text) {
     this.setState({ sql: text });
   }
 
   handleSubmit() {
-    this.setState({
-      response: undefined,
-      loading: true,
-      error: undefined
-    });
+    const { sql } = this.state;
+    const key = ++this.uniqueKey;  // eslint-disable-line
```

Comment:
I'd avoid exceptions and allow it in `.eslintrc.js`

Reply:
done in 71c1404